core/state_processor.go

```diff
@@ -129,9 +134,15 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, gp *GasPool
 	receipt.TxHash = tx.Hash()
 	receipt.GasUsed = result.UsedGas
 
+	if msg.IsDepositTx() && config.IsOptimismRegolith(evm.Context.Time) {
+		// The actual nonce for deposit transactions is only recorded from Regolith onwards.
+		// In bedrock, DepositNonce must remain nil
```

Comment:
```suggestion
		// Before the Regolith fork the DepositNonce must remain nil
```

I think we should communicate Regolith abstractly as part of Bedrock; on mainnet we plan to activate it from the start. So let's avoid saying "in bedrock" when we mean pre-regolith.